src/Configuration/TargetUrl.php

```diff
@@ -32,4 +32,17 @@ public function getHeaders() {
         return $this->headers;
     }
 
+    public static function fromString($url) {
```

Comment:
You need some additional checks here, to make sure, that the components are really filled. But I like that method :-)